### ISSUE TITLE
AE2 support

### ISSUE THREAD

**Issue by tuturox91** [AlmostReliable/ponderjs#7]
### Version

ponderjs-1.18.2-1.1.8 and appliedenergistics2-11.1.2

### Describe the bug!

AE 2 cables can't show in ponder(

![image](https://user-images.githubusercontent.com/51265345/185212982-97fcae98-73b0-481b-a5cd-9a256295c152.png)
![image](https://user-images.githubusercontent.com/51265345/185213019-8ef029a0-c5fd-4b6a-b49b-2be1256ff533.png)

### Additional Context

Yes

### Did the issue happen in singleplayer or on a server?

Singleplayer

**Comment by Ludi87** [comment]
If you include the cables in the structure nbt they can show, but they dont connect :/ not sure if there's a way to fix that..?

![image](https://user-images.githubusercontent.com/13112051/185855682-1846a8fa-a6ad-4cee-a257-8f18b85f05e5.png)

**Comment by LLytho** [comment]
Not sure if this is something related to create and ponder. Need to figure it out. And for connecting you probably need to setup the blockstates correctly (if they use it).

**Comment by Ludi87** [comment]
The blockstates are set correctly, but with an NBT Editor I saw that part of the NBT wasn't saved with the schematic (the "extra center" bit). I imagine that's what's setting the connection to other ones.
I couldn't figure out how to set NBT so the quotation marks show up, since you couldn't do something like
```js
scene.world.modifyTileNBT([1, 1, 3], (nbt) => {
    nbt."Extra:Center": {
        gn: {
            p: 0,
            k: "-1L",
            g: "2L"
        }
    }
}
```

![image](https://user-images.githubusercontent.com/13112051/185917780-64edf221-016c-478c-93ce-c8a9cd70076e.png)

**NBT** of a cable in the structure:
![image](https://user-images.githubusercontent.com/13112051/185918508-9d94636f-122d-4c83-b0ea-e610d322d90c.png)

![image](https://user-images.githubusercontent.com/13112051/185918572-8fc55840-e053-45a2-adc4-c1025e6362d1.png)

also tried this, no effect:
```js
                scene.world.modifyTileNBT([1, 1, 2], (nbt) => {
                    nbt = [
                        {
                            id: "ae2:cable_bus",
                            "item:center": "ae2:fluix_glass_cable",
                            "extra:center":{
                                gn:{
                                    p: 0,
                                    k: "-1L",
                                    g: "2L"
                                }
                            }
                        },
                    ];
                });

                scene.world.modifyTileNBT([1, 1, 1], (nbt) => {
                    nbt = [
                        {
                            id: "ae2:cable_bus",
                            "item:center": "ae2:fluix_glass_cable",
                            "extra:center":{
                                gn:{
                                    p: 0,
                                    k: "-1L",
                                    g: "2L"
                                }
                            }
                        },
                    ];
                });
```

**Comment by tuturox91** [comment]
If you find how to fix this. This is be awesome! The ae2 has no guide book. And such a visual tutorial would be very useful in modpacks!

**Comment by rlnt** [comment]
We'll leave this open in case someone finds out a way to do it.

**Comment by rlnt** [comment]
One of the AE2 devs looked into this and it was indeed not possible with the previous version of Applied Energistics. But they introduced a way to make AE objects work within Ponder.

The tracking PR can be found here: https://github.com/AppliedEnergistics/Applied-Energistics-2/pull/6661

I'll close this issue when it's ready.

**Comment by shartte** [comment]
https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/master/ponderjs.md

<blockquote><img src="https://opengraph.githubassets.com/6a102ac0cee48ac9cf91e4783a8c60766d5d02405873ea771fd86e87dab01be6/AppliedEnergistics/Applied-Energistics-2" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/AppliedEnergistics/Applied-Energistics-2">Applied-Energistics-2/ponderjs.md at master · AppliedEnergistics/Applied-Energistics-2</a></strong></div><div>A Minecraft Mod about Matter, Energy and using them to conquer the world.. - Applied-Energistics-2/ponderjs.md at master · AppliedEnergistics/Applied-Energistics-2</div></blockquote>

**Comment by rlnt** [comment]
Implemented here and ported to all 1.18+ versions: https://github.com/AppliedEnergistics/Applied-Energistics-2/pull/6662

The guide page can be found [here](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/master/ponderjs.md) and will be also available on our [wiki](https://github.com/AlmostReliable/ponderjs-forge/wiki).

**Comment by tuturox91** [comment]
Wow this is amazing!
